### PR TITLE
Datahub - add info when metadata has no online ressource

### PR DESCRIPTION
**Pull request by rcaplier**
### Description

This PR introduces a new warning block for when a record dataset has neither API, download or other links to display.

### Screenshots

image

### Quality Assurance Checklist

- [ ] Commit history is devoid of any _merge commits_ and readable to facilitate reviews
- [x] If **new logic** ⚙️ is introduced: unit tests were added
- [x] If **new user stories** 🤏 are introduced: E2E tests were added
- [ ] If **new UI components** 🕹️ are introduced: corresponding stories in Storybook were created
- [ ] If **breaking changes** 🪚 are introduced: add the `breaking change` label
- [ ] If **bugs** 🐞 are fixed: add the `backport <release branch>` label
- [ ] The [documentation website](docs) 📚 has received the love it deserves

---

**This work is sponsored by Swisstopo geocat**.

### CODE REVIEW

File: libs/ui/elements/src/lib/search-results-error/search-results-error.component.html [hunk 1]
```diff
@@ -13,14 +16,29 @@
...
     <div translate>search.error.couldNotReachApi</div>
   </div>
-  <div *ngIf="type === types.RECEIVED_ERROR">
+  <div *ngIf="type === types.NO_LINK" data-test="no-link-error">
+    <div class="relative opacity-40">
+      <mat-icon class="material-symbols-outlined face">computer</mat-icon>
+      <mat-icon class="material-symbols-outlined question-mark1"
+        >question_mark</mat-icon
+      >
+      <mat-icon class="material-symbols-outlined question-mark2"
```
Comment (cmoinier) [inline libs/ui/elements/src/lib/search-results-error/search-results-error.component.html:25]: What are these two question marks icons for ? They are not in the screenshot you provided.

File: translations/fr.json [hunk 2]
```diff
@@ -332,6 +332,7 @@
   "search.error.couldNotReachApi": "Problème de connexion à l'API",
   "search.error.receivedError": "Erreur retournée",
   "search.error.recordNotFound": "Cette donnée n'a pu être trouvée.",
+  "search.error.nolink": "Ce dataset n'a pas encore de lien, réessayez plus tard s'il vous plaît.",
```
Comment (cmoinier) [inline translations/fr.json:335]: I would've made "liens" plural :slightly_smiling_face:

**Review by cmoinier — CHANGES_REQUESTED** [review]
Thanks for this feature @rcaplier ! :smiley: It looks really good already. Two remarks/questions under here + would be nice to add e2e testing (difficult to do for other errors, but for the absence of links, you would just have to point to a dataset where you would have removed all the links, perhaps there is already an existing one that you used for this PR? LMK if you need help with that).

File: translations/fr.json (hunk 2 above)
Comment (cmoinier) [reply translations/fr.json:335]: Mmmh seen like that, yes it makes sense for the singular, my bad!

File: libs/ui/elements/src/lib/search-results-error/search-results-error.component.html (hunk 1 above)
Comment (rcaplier) [reply libs/ui/elements/src/lib/search-results-error/search-results-error.component.html:25]: Wrong copy/pasta. They don't even display 😆 
Fixed it. Good catch !

**Comment by rcaplier** [comment]
> Thanks for this feature @rcaplier ! 😃 It looks really good already. Two remarks/questions under here + would be nice to add e2e testing (difficult to do for other errors, but for the absence of links, you would just have to point to a dataset where you would have removed all the links, perhaps there is already an existing one that you used for this PR? LMK if you need help with that).

Thank you. I've had to make a new dataset with no link in it to see the result. Do I have to make another dump so we have it and point to it in the e2e tests ? Or maybe I can cheat in the e2e by simulating a record with no link ? 😎

**Comment by cmoinier** [comment]
> > Thanks for this feature @rcaplier ! 😃 It looks really good already. Two remarks/questions under here + would be nice to add e2e testing (difficult to do for other errors, but for the absence of links, you would just have to point to a dataset where you would have removed all the links, perhaps there is already an existing one that you used for this PR? LMK if you need help with that).
> 
> Thank you. I've had to make a new dataset with no link in it to see the result. Do I have to make another dump so we have it and point to it in the e2e tests ? Or maybe I can cheat in the e2e by simulating a record with no link ? 😎

Yes you can use that one! I think simulating a record without any links would be too time consuming, I would use the one you got :wink:

**Review by jahow — COMMENTED** [review]
Just a general comment, could we maybe be more specific and instead of "noLink" use "recordHasNoLink"? E.g. `search.error.RecordHasNoLink`, `ErrorType.RECORD_HAS_NO_LINK`, etc.

**Review by cmoinier — APPROVED** [review]
Thanks for the last changes, LGTM :ok_hand:

**Pull request merged**